### PR TITLE
test: adding one more watermark test

### PR DESCRIPTION
**Pull request by alyssawilk**
*Risk Level*: n/a (test only)
*Testing*: 5k runs cleanly
*Docs Changes*: n/a
*Release Notes*: n/a

### CODE REVIEW

**Comment by alyssawilk** [comment]
Alternately if we don't think a random-backup test is worth the extra test code, I can just remove the TODO.

**Comment by enzoferrari** [comment]
> Alternately if we don't think a random-backup test is worth the extra test code, I can just remove the TODO.

Thx so much

File: test/integration/http2_upstream_integration_test.cc [hunk 1]
```diff
@@ -280,6 +280,15 @@ TEST_P(Http2UpstreamIntegrationTest, ManyLargeSimultaneousRequestWithBufferLimit
   manySimultaneousRequests(1024 * 20, 1024 * 20);
 }
 
+TEST_P(Http2UpstreamIntegrationTest, ManyLargeSimultaneousRequestWithRandomBackup) {
+  config_helper_.addFilter(R"EOF(
+  name: random-pause-filter
+  config: {}
+  )EOF");
+
+  manySimultaneousRequests(1024 * 20, 1024 * 20);
```
Comment (mattklein123) [inline test/integration/http2_upstream_integration_test.cc]: If we are going to do a random test, should we do this multiple times? Perhaps 5 or 10? WDYT?

File: test/integration/filters/random_pause_filter.cc [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+// randomly.
+class RandomPauseFilter : public PassThroughFilter {
+public:
+  RandomPauseFilter(absl::Mutex& rand_lock, TestRandomGenerator& rng)
+      : rand_lock_(rand_lock), rng_(rng) {}
+
+  Http::FilterDataStatus encodeData(Buffer::Instance& buf, bool end_stream) override {
+    absl::WriterMutexLock m(&rand_lock_);
+    uint64_t random = rng_.random();
+    if (random % 5 == 0) {
```
Comment (mattklein123) [inline test/integration/filters/random_pause_filter.cc]: Just curious how you picked this? Maybe a comment?

File: test/integration/http2_upstream_integration_test.cc (hunk 1 above)
Comment (alyssawilk) [reply test/integration/http2_upstream_integration_test.cc]: I'm fine with it being fuzzed over many CI runs, but if you think it's worth it I can refactor all the initialization set up out of the function so we can do the requests multiple times.

Comment (mattklein123) [reply test/integration/http2_upstream_integration_test.cc]: Up to you, if you think CI is enough that's fine.

**Pull request merged**